### PR TITLE
[MPS] Fix base shape size for view ops in case of multiple slices

### LINKED ISSUE #84364: [MPS] Indexing into new axis of tensor causes its every element to become ±Inf
### 🐛 Describe the bug

Minimal repro of a situation that occurs [inside k-diffusion](https://github.com/crowsonkb/k-diffusion/blob/4567328deec0f90a549dde53c1858cf7ed0140dd/k_diffusion/utils.py#L43), when any sampler runs `to_d()` on a sigma beyond the first one.  
The [`[None]` indexing convention](https://numpy.org/doc/stable/user/basics.indexing.html#dimensional-indexing-tools) is used to add a dimension to a tensor.

Works fine on-CPU:

```python
torch.tensor(1., device='cpu') / torch.tensor([2., 2.], device='cpu')[0][None]
# tensor([0.5000])

torch.tensor(1., device='cpu') / torch.tensor([2., 2.], device='cpu')[1][None]
# tensor([0.5000])
```

Breaks for indices > 0 on MPS:

```python
# good
torch.tensor(1., device='mps') / torch.tensor([2., 2.], device='mps')[0][None]
# tensor([0.5000], device='mps:0')

# BAD!
torch.tensor(1., device='mps') / torch.tensor([2., 2.], device='mps')[1][None]
# tensor([inf], device='mps:0')
```

The consequence is that black images are generated:

You can work around it by detaching the tensor:

```python
torch.tensor(1., device='mps') / torch.tensor([2., 2.], device='mps')[1][None].detach()
# tensor([0.5000], device='mps:0')
```

### Versions

PyTorch version: 1.13.0.dev20220606
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.5 (arm64)
GCC version: Could not collect
Clang version: 13.0.0 (clang-1300.0.29.30)
CMake version: version 3.22.1
Libc version: N/A

Python version: 3.9.12 (main, Jun  1 2022, 06:34:44)  [Clang 12.0.0 ] (64-bit runtime)
Python platform: macOS-12.5-arm64-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.0rc2
[pip3] torch==1.13.0.dev20220606
[pip3] torchaudio==0.14.0.dev20220603
[pip3] torchvision==0.14.0a0+f9f721d
[conda] numpy                     1.23.0rc2                pypi_0    pypi
[conda] torch                     1.13.0.dev20220606          pypi_0    pypi
[conda] torchaudio                0.14.0.dev20220603          pypi_0    pypi
[conda] torchvision               0.14.0a0+f9f721d          pypi_0    pypi

cc @kulinseth @albanD

### PR DESCRIPTION
**Pull request by DenisVieriu97**
Fixes https://github.com/pytorch/pytorch/issues/84364, https://github.com/pytorch/pytorch/issues/85592

Fixes bug for view ops where the base shape would be incorectly determined.
E.g for the following tensor `torch.tensor([0.5, 0.5], device="mps")[1][None]`, we could consider the base shape of the parent tensor as 1, while the actual base shape is 2.

### CODE REVIEW

**Review by kulinseth — APPROVED** [review]
Looks good.
